### ISSUE TITLE
[Question]: Context retriever and memory

### ISSUE THREAD

**Issue by grauvictor** [run-llama/llama_index#8911]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I have a question about `ContextChatEngine` and more precisely the `chat` method. In this method the context nodes are retrieved with:
```
 def chat(
        self, message: str, chat_history: Optional[List[ChatMessage]] = None
    ) -> AgentChatResponse:
        if chat_history is not None:
            self._memory.set(chat_history)
        self._memory.put(ChatMessage(content=message, role="user"))

        context_str_template, nodes = self._generate_context(message)
        ...
```

The memory does not seem to be used to find the context. 

So, for example, in conversation:

human: "I'd like to make a tiramisu"
bot: bot answer
human: "what's the recipe?

Can you confirm that the context retrieved for the last input ("What's the recipe") does not take memory into account, and that the nodes retrieved from the knowledge base will therefore have no direct link with the first input ("I'd like to make a tiramisu")? 

Is this intended behaviour? 

Thank you in advance for your reply

**Comment by gich2009** [comment]
Hi @grauvictor, I reported a similar issue to this when the library was upgraded post 0.8.56. The context chat engine was actually taking the memory into account at that time but after the upgrade, context keeping was not as good. 

I have experimented and realized that the condense_question chat engine is good at keeping context and retrieving information but it is very poor at adhering to the system prompt whereas the context chat engine is good at retrieving information and adhering to the system prompt but it is poor at keeping context.
